src/lib.rs

```diff
@@ -1,23 +1,23 @@
 //! Lazily evaluated monadic IO actions.
 
-use std::{ io::{ Read, Result, Write }, path::Path, mem, fs };
+use std::{ io::{ Read, Result, Write }, path::Path, mem, fs, marker::PhantomData };
 use Inner::{ Eval, Pure, Done };
 
 /// A chain of lazy, composable IO actions.
-pub struct IO <O> (Inner<O>);
+pub struct IO <'a, O = ()> (Inner<'a, O>);
 
-enum Inner <O> {
-    Eval (Box<dyn Action<Output = O>>),
+enum Inner <'a, O> {
```

Comment:
Expended is a hard word. It's not that complicated,but at least for me it's not in the "L1 brain word cache" which does make reading it a bit less smooth, I like the comments and don't really have an opinion on Eval -> Chain but I think end is a better name than expanded

This may be my 1am brain speaking

Reply:
What about `Depleted`?

Reply:
Doesn't feel like it's any different for me tbh?

Reply:
After IRL discussion we agreed to keep Depleted as the name, The descriptiveness is better than using simple words